contracts/dev-utils/contracts/src/OrderValidationUtils.sol

```diff
@@ -0,0 +1,190 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-exchange/contracts/src/interfaces/IExchange.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibOrder.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibMath.sol";
+import "@0x/contracts-utils/contracts/src/LibBytes.sol";
+import "@0x/contracts-asset-proxy/contracts/src/libs/LibAssetData.sol";
+
+
+contract OrderValidationUtils is
+    LibAssetData,
+    LibMath
+{
+    using LibBytes for bytes;
+
+    // solhint-disable var-name-mixedcase
+    IExchange internal EXCHANGE;
+    bytes internal ZRX_ASSET_DATA;
+    address internal ERC20_PROXY_ADDRESS;
+    // solhint-enable var-name-mixedcase
+
+    constructor (address _exchange, bytes memory _zrxAssetData)
+        public
+    {
+        EXCHANGE = IExchange(_exchange);
+        ZRX_ASSET_DATA = _zrxAssetData;
+        ERC20_PROXY_ADDRESS = EXCHANGE.getAssetProxy(ERC20_PROXY_ID);
+    }
+
+    /// @dev Fetches information for order and maker/taker of order.
+    /// @param order The order structure.
+    /// @param signature Proof that order has been created by maker.
+    /// @return OrderInfo, the remaining amount fillable by the taker, and validity of signature for given order.
+    function getOrderRelevantState(LibOrder.Order memory order, bytes memory signature)
+        public
+        view
+        returns (
+            LibOrder.OrderInfo memory orderInfo,
+            uint256 fillableTakerAssetAmount,
+            bool isValidSignature
+        )
+    {
+        // Get info specific to order
+        orderInfo = EXCHANGE.getOrderInfo(order);
+
+        // Validate the maker's signature
+        // If the signature does not need to be validated, `0x01` can be supplied for the signature to always return `false`.
+        address makerAddress = order.makerAddress;
+        isValidSignature = EXCHANGE.isValidSignature(
+            orderInfo.orderHash,
+            makerAddress,
+            signature
+        );
+
+        // Get the transferable amount of the `makerAsset`
+        uint256 transferableMakerAssetAmount = getTransferableAssetAmount(order.makerAssetData, makerAddress);
+
+        // Assign to stack variables to reduce redundant mloads
+        uint256 takerAssetAmount = order.takerAssetAmount;
+        uint256 makerFee = order.makerFee;
+    
+        // Get the amount of `takerAsset` that is purchasable given the transferability of `makerAsset` and `makerFeeAsset`
+        uint256 purchasableTakerAssetAmount;
+        if (order.makerAssetData.equals(ZRX_ASSET_DATA)) {
```

Comment:
Why are we still special-casing ZRX here? I thought the fee can be paid in any token now?

Reply:
This is still in the `development` branch so that you can start using it with 2.0 immediately, but it's written in a way that makes it very easy to update to 3.0.